qtrees/data_processor.py

```diff
@@ -0,0 +1,311 @@
+import pandas as pd
+import numpy as np
+from functools import reduce
+import datetime as dt
+from typing import Union, Optional
+from sklearn.preprocessing import OrdinalEncoder
+
+ROLLING_WINDOW = 7
+FORECAST_HORIZON = 14
+AUTOREGRESSIVE_LAG = 3
+
+class Data_loader:
+    def __init__(self, engine):
+        self.engine = engine
+
+    def download_nowcast_inference_data(self, date: dt.date = dt.date.today() - dt.timedelta(days=1), public_run: bool = False, batch_size: Optional[int] = None, batch_num: Optional[int] = None):
+        # TODO: Figure out a way to download data in batches s.t. we do not have to hold that much info in memory
+        self.forecast = False
+        self.date = date
+        self.public_run = public_run
+        self.batch_size = batch_size
+        self.batch_num = batch_num
+        data = self._download_data(with_sensors=False)
+        if data is not None:
+            data = data.merge(self._get_weather_measurements(), how="left", left_on="timestamp", right_index=True)
+        return data
+    
+    def download_forecast_inference_data(self, date: dt.date = dt.date.today(), public_run: bool = False, batch_size: Optional[int] = None, batch_num: Optional[int] = None):
+        self.forecast = True
+        self.date = date
+        self.public_run = public_run
+        self.batch_size = batch_size
+        self.batch_num = batch_num
+        data = self._download_data(with_sensors=False)
+        return data
+    
+    def download_nowcast_training_data(self, public_run: bool = False):
```

Comment:
download_nowcast_training_data und download_forecast_training_data sind ja bis auf einen parameter gleich. Könnte auch in ne Funktion ausgelagert werden das was gleich ist.

Reply:
true